src/ssh/sftp_dir_iterator.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2019-2022 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <multipass/ssh/sftp_dir_iterator.h>
+
+#include <multipass/ssh/sftp_utils.h>
+
+#include <multipass/format.h>
```

Comment:
Unless there is some issue on the order of `#include`'s, could you please group the `multipass` includes together, then a blank line and then the `stdexcept`?